src/ghdcbot/adapters/discord/api.py

```
@@ -192,28 +192,125 @@ def remove_role(self, discord_user_id: str, role_name: str) -> None:
         if response.status_code not in (200, 201):
             self._logger.warning(
-                "Discord send_message failed",
+                "Discord post_channel_message failed",
                 extra={"channel_id": channel_id, "status_code": response.status_code},
             )
+            return None
+
+        try:
+            data = response.json()
+            return str(data.get("id") or "")
+        except Exception:
```

Ruff (0.16.2)

[warning] Do not catch blind exception: `Exception`

(BLE001)